### ISSUE TITLE
Add feature to dump rewritten formula

### ISSUE THREAD

**Issue by 4tXJ7f** [CVC4/CVC4#2702]
Without doing equisatisfiable rewrites.

**Comment by 4tXJ7f** [comment]
For context, here's the email that prompted this issue:
```
I was looking for a tool to minimize a SMT2-Problem. The problem is 
computer generated and has a lot of noise in it like "(>= a 1) (<= a 1)" 
(which is just a constant and could be replaced with 1 in the rest of 
the problem) or things like "(>= 11 (+ 0 0 0 0 0 12 (- 4 3))" which is 
just true etc.

Some background: we have a tool which solves some huge problems by goal 
programming with SMT2. Every time we have reached a new goal, we could 
then reduce the smt-file by inserting what we have learned (stronger 
boundries on int vars mainly). Because this is done in parallel and 
orchestrated, it would be nice to have another process, which just 
minimizes the problem and feeds the smaller problem back into the loop. 
Simply by removing variables known to be fixed numbers would speed up 
things because there are a lot of variables in a solution and solutions 
are passed around the processes (and problems too). We use the whole 
thing successfully to calculate big duty roasters of our clients 
(www.planik.ch).

Somebody from ddSMT kindly guided me to CVC4. I was not aware that CVC4 
could do that. I was told to use "cvc4 --dump=assertions:post-everything 
--preprocess-only -qq" and indeed, it works ... almost.

We only do problems of the type QF_LIA. When I give CVC4 with above 
options this smt-file:

--snip--
(set-option :print-success false)
(set-option :produce-models true)
(set-logic QF_LIA)
(declare-fun a () Int)
(assert ...)
(assert (= a 0))
(check-sat)
--snap--

.. it gives me this:

--snip--
(set-option :incremental false)
(set-option :print-success false)
(set-option :produce-models true)
(set-info :notes "CVC4 always dumps the most general, all-supported 
logic (below), as some internals might require the use of a logic more 
general than the input.")
(set-logic ALL)
(declare-fun a () Int)
(assert ...)
(assert true)
(assert true)
(check-sat)
unknown
--snap--

Which is, in my opinion, not quite the same: in the minimized version, 
variable "a" could be any number whereas in the original it can only be 0.

Now my question:

- Is there a way to tell cvc4 to print out variables with a fixed range 
too, e.g. to keep variable "a" set to 0?
- if no, with which option(s) can "a" somehow be removed from the 
declaration part in the output (=totally removed in the output)?

I tried a lot of options and read the help, but I don't understand all 
the theoretical terms. Any help would be apriciated.
```
